src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final double defaultRadius = 10;
+    private static final String defaultColor = "White";
+    private ColorSupplier colorSupplier;
+    private Random random = new Random();
+
+    public FigureSupplier() {
+        this.colorSupplier = new ColorSupplier();
+    }
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(5);
+        double randomFrom = 1;
+        double randomTo = 10;
+        double side1 = random.nextDouble(randomFrom,randomTo);
+        double side2 = 0;
+        if (figureNumber >= 2) {
+            side2 = random.nextDouble(randomFrom, randomTo);
+        }

```

Comment:
kind of overcomplicated. If you want to use local variable, next time consider using `() -> random.nextDouble(randomFrom,randomTo)` Supplier, if you know what it is.